generator/src/test/java/com/scottlogic/deg/generator/cucumber/steps/DateValueStep.java

```diff
@@ -4,8 +4,9 @@
 import cucumber.api.java.en.When;
 
 public class DateValueStep {
-
-    public static final String DATE_REGEX = "((\\d{4})-(\\d{2})-(\\d{2}T(\\d{2}:\\d{2}:\\d{2}\\.\\d{3})))$";
+    // matches ISO-8601; could be more specific.
+    // we use a plain .* at the end of the time bit because if the earlier stuff is matched then we don't really need to be stringent about the rest of it
+    public static final String DATE_REGEX = "((\\d{4})-(\\d{2})-(\\d{2}T(\\d{2}:\\d{2}:\\d{2}\\.\\d{3}.*)))$";
```

Comment:
The cucumber framework should ensure that dates are correctly formatted.

Is there the ability to share the format (or regex of the format) from `AtomicConstraintReaderLookup` to ensure cucumber accepts the same formats as production? 

If not can we update `.*` to `((+\d{2})|Z)?` to catch any formatting errors earlier.

Reply:
IIRC, the code above is purely for determining the test writer's intent, I don't think it's the place for ensuring that the generator correctly parses or displays dates. In fact, to do the latter at all, we _need_ the ability for incorrectly formatted dates to get put into `{ "date": ... }` structures, so the validation here can't be completely strict.

Reply:
I think it would be better if it changed, but any mistakes i _think_ would be picked up correctly.